### PR TITLE
[Core] Remove node_erase_process.h, this is a 4 years old TODO

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

Basically that

**🆕 Changelog**

- [Remove node_erase_process.h](https://github.com/KratosMultiphysics/Kratos/commit/e6d050ef24843db6026ace6e04d12d45c88c3284)
- [Refactor meshing app](https://github.com/KratosMultiphysics/Kratos/commit/a9ba52b448cdc1114b724634399ebbeee8bcb7d2)
- [Unused includes](https://github.com/KratosMultiphysics/Kratos/commit/6a715d1a040a557e7245d4985aad34bb06635fbc)

### CODE REVIEW

File: applications/MeshingApplication/custom_utilities/local_refine_geometry_mesh.hpp [hunk 1]
```diff
@@ -33,7 +33,7 @@
 #include "includes/constitutive_law.h"
 #include "geometries/geometry.h"
 #include "utilities/math_utils.h"
-#include "processes/node_erase_process.h"
+#include "processes/entity_erase_process.h"
```
Comment (philbucher) [inline applications/MeshingApplication/custom_utilities/local_refine_geometry_mesh.hpp]: move to cpp

File: applications/MeshingApplication/external_includes/tetgen_pfem_contact.h [hunk 2]
```diff
@@ -30,7 +30,7 @@
 #include "geometries/triangle_3d_3.h"
 #include "geometries/tetrahedra_3d_4.h"
 #include "meshing_application_variables.h"
-#include "processes/node_erase_process.h"
+#include "processes/entity_erase_process.h"
```
Comment (philbucher) [inline applications/MeshingApplication/external_includes/tetgen_pfem_contact.h]:
```suggestion
```

unused, at least in this file

File: kratos/python/add_processes_to_python.cpp [hunk 3]
```diff
@@ -235,8 +234,7 @@ void  AddProcessesToPython(pybind11::module& m)
     .def(py::init<ModelPart&, std::size_t>())
     ;
 
-//     py::class_<EntitiesEraseProcess<Node<3>>, EntitiesEraseProcess<Node<3>>::Pointer, Process>(m,"NodeEraseProcess") // TODO: Replace when the remainings of NodeEraseProcess have been cleaned up
-    py::class_<NodeEraseProcess, NodeEraseProcess::Pointer, Process>(m,"NodeEraseProcess")
+    py::class_<EntitiesEraseProcess<Node<3>>, EntitiesEraseProcess<Node<3>>::Pointer, Process>(m,"NodeEraseProcess")
     .def(py::init<ModelPart&>())
```
Comment (philbucher) [inline kratos/python/add_processes_to_python.cpp:238]: please issue a deprecation warning telling that the `EntitiesEraseProcess` should be used instead (it is exposed to python I assume?)

**Review by philbucher — CHANGES_REQUESTED** [review]
minor comments otherwise :+1:

File: kratos/python/add_processes_to_python.cpp (hunk 3 above)
Comment (loumalouomega) [reply kratos/python/add_processes_to_python.cpp:238]: Well, it does the same, and the TODO has been there 4 years

File: applications/MeshingApplication/custom_utilities/local_refine_geometry_mesh.cpp [hunk 4]
```diff
@@ -17,8 +17,12 @@
 #include <unordered_map>
 
 // External includes
+#ifdef _OPENMP
+#include <omp.h>
+#endif
```
Comment (philbucher) [inline applications/MeshingApplication/custom_utilities/local_refine_geometry_mesh.cpp]:
```suggestion
```

I am pretty sure this is unused

File: kratos/python/add_processes_to_python.cpp (hunk 3 above)
Comment (philbucher) [inline kratos/python/add_processes_to_python.cpp:238]:
```suggestion
```

well if it is so long up for removal then remove it I suggest

**Comment by philbucher** [comment]
Why did you revert?

**Comment by loumalouomega** [comment]
Because I think it could be useful to be exposed in python with the new
interface (not the old one)

**Review by philbucher — APPROVED** [review]

**Pull request merged**